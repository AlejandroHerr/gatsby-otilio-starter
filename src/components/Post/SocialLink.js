import React from 'react';
import PropTypes from 'prop-types';

import SimpleIcon from '../icons/SimpleIcon';

import styles from './Post.module.scss';

const SocialLink = ({ service, url }) => (
  <a className={styles.post_author__social} href={url}>
    <SimpleIcon brand={service.toLowerCase()} className={styles.post_author__social_icon} />
  </a>
);

SocialLink.propTypes = {
  service: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default SocialLink;
