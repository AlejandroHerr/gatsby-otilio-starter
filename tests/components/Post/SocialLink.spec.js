import React from 'react';
import renderer from 'react-test-renderer';

import SocialLink from '../../../src/components/Post/SocialLink';

const defaultProps = {
  service: 'Test_Service',
  url: 'test:url',
};

describe('Post', () => {
  describe('SocialLink', () => {
    it('should match the exact snapshot', () => {
      const tree = renderer.create(<SocialLink {...defaultProps} />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
