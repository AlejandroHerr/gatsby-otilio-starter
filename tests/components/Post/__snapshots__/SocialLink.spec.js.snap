// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post SocialLink should match the exact snapshot 1`] = `
<a
  className="post_author__social"
  href="test:url"
>
  <svg
    className="simple_icon simple_icon__test_service post_author__social_icon"
    style={Object {}}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path />
  </svg>
</a>
`;
